src/scripts/DataLayerUtils.js

```diff
@@ -48,6 +48,33 @@ const utils = {
    */
   isObject: function(obj) {
     return (obj && typeof obj === 'object' && !Array.isArray(obj));
+  },
+
+  /**
+   * Checks if the given object has the nested property.
+   *
+   * @param {String} propertyPath The property to be be checked
+   * @param {Object} obj The object that will be checked.
+   * @returns {Boolean} true if the given object has the nested property, false otherwise.
+   */
+  hasOwnNestedProperty: function(propertyPath, obj) {
```

Comment:
we may want to use the lodash `has` function instead of having our own implementation:
https://lodash.com/docs#has